javascript-view-swapping/main.js

```diff
@@ -0,0 +1,25 @@
+const tabContainer = document.querySelector('.tab-container');
+const tab = document.querySelectorAll('.tab');
+const view = document.querySelectorAll('.view');
+
+function handleClick(event) {
+  if (!event.target.matches('.tab')) {
+    return;
+  }
+  for (let i = 0; i < tab.length; i++) {
+    if (tab[i] === event.target) {
+      tab[i].className = 'tab active';
+      const dataView = event.target.getAttribute('data-view');
+      for (let j = 0; j < view.length; j++) {
+        if (view[j].getAttribute('data-view') === dataView) {
+          view[j].className = 'view';
+        } else {
+          view[j].className = 'view hidden';
+        }
+      }
```

Comment:
Take these lines out of the outer `for` loop and insert it after line 22. 
Having nested `for` loops should be avoided if at all possible as it is much slower as far as execution time.